src/platform/user/profile/vet360/actions/transactions.js

```diff
@@ -25,151 +26,197 @@ export const VET360_TRANSACTION_UPDATE_FAILED =
         });
-      } else {
-        dispatch({
-          type: VET360_TRANSACTION_UPDATED,
-          transaction: transactionRefreshed,
-        });
-
-        if (isFailedTransaction(transactionRefreshed) && analyticsSectionName) {
-          recordEvent({
-            event: 'profile-edit-failure',
-            'profile-action': 'save-failure',
-            'profile-section': analyticsSectionName,
-          });
-        }
       }
-    } catch (err) {
-      dispatch({
-        type: VET360_TRANSACTION_UPDATE_FAILED,
-        transaction,
-        err,
-      });
     }
+  } catch (err) {
+    dispatch({
+      type: VET360_TRANSACTION_UPDATE_FAILED,
+      transaction,
+      err,
+    });
+  }
+};
+
+export const createTransaction = (
+  route,
+  method,
+  fieldName,
+  payload,
+  analyticsSectionName,
+) => async dispatch => {
+  const options = {
+    body: JSON.stringify(payload),
+    method,
+    headers: {
+      'Content-Type': 'application/json',
+    },
   };
-}
 
-export function createTransaction(
+  try {
+    dispatch({
+      type: VET360_TRANSACTION_REQUESTED,
+      fieldName,
+      method,
+    });
+
+    const transaction = isVet360Configured()
+      ? await apiRequest(route, options)
+      : await localVet360.createTransaction();
+
+    recordEvent({
+      event: method === 'DELETE' ? 'profile-deleted' : 'profile-transaction',
+      'profile-section': analyticsSectionName,
+    });
+
+    dispatch({
+      type: VET360_TRANSACTION_REQUEST_SUCCEEDED,
+      fieldName,
+      transaction,
+    });
+  } catch (error) {
+    dispatch({
+      type: VET360_TRANSACTION_REQUEST_FAILED,
+      error,
+      fieldName,
+    });
+  }
+};
+
+export const validateAddress = (
   route,
   method,
   fieldName,
   payload,
   analyticsSectionName,
-) {
-  return async dispatch => {
-    const options = {
-      body: JSON.stringify(payload),
-      method,
-      headers: {
-        'Content-Type': 'application/json',
-      },
-    };
-
-    try {
-      dispatch({
-        type: VET360_TRANSACTION_REQUESTED,
-        fieldName,
-        method,
-      });
-
-      const transaction = isVet360Configured()
-        ? await apiRequest(route, options)
-        : await localVet360.createTransaction();
+) => async dispatch => {
+  const addressPayload = { address: { ...payload } };
+
+  const options = {
+    body: JSON.stringify(addressPayload),
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  };
 
-      recordEvent({
-        event: method === 'DELETE' ? 'profile-deleted' : 'profile-transaction',
-        'profile-section': analyticsSectionName,
+  try {
+    const response = isVet360Configured()
+      ? await apiRequest('/profile/address_validation', options)
+      : await localVet360.addressValidationSuccess(payload);
+
+    const { addresses } = response;
+
+    addresses.filter(
+      address =>
+        get(address, 'address_meta_data.delivery_point_validation') !==
```

Comment:
> We are moving away from using lodash and our own custom `get` helper (
> 
> https://github.com/department-of-veterans-affairs/vets-website/blob/6a27450abed0eb724d9fa0334633ff8ffe700f78/src/platform/utilities/data/get.js#L16
> 
> We are moving away from using lodash and our own custom get helper (

Why are we moving away from Lodash ?? I have only heard good things about it and in most cases the the benefits outweigh this disadvantage

Is there a valid reason for the move ?

Reply:
To be clear, we are trying to use the standard ES6 optional chaining, _not_ our custom `get` function. The general consensus is that if functionality is available in native JS we should use that vs using a 3rd party utility.

I wasn't around when the custom `get` function was created, but I believe the reason for creating it in the first place was that it was lighter weight than lodash's functional programming version of `get`. My understanding is that you can't include lodash fp helpers on a la carte; you have to include the entire lodash fp library, which is one of the largest dependencies we have. But @cvalarida or @rianfowler might have more context on that particular issue.

Reply:
If instead of Lodash we use the ES6 build in function then I am all for it, but it did not made sense to me to use a custom function over Lodash unless there was a really good use case that proof that a custom function will outweigh the benefits of using a library like Lodash

In most cases this is true
Library best than Custom
Build in best than Library

I am all for using build-in functions instead of libraries.

Reply:
I think we agreed with that as well, Joe. I brought up this question earlier this month https://dsva.slack.com/archives/C5HP4GN3F/p1572621281246200

Reply:
For complete clarity here, the standard is now to use optional chaining, as described here - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining.

So, this -

```js
get(address, 'address_meta_data.delivery_point_validation')
```

Becomes -

```js
address?.address_meta_data?.delivery_point_validation
```

So, we don't need any function at all. The standard in vets-website is generally to prefer ES6 over Lodash unless there is clear case for using Lodash. My personal opinion is that the ES6 version is more elegant and doesn't include the overhead of Lodash in our resultant JavaScript bundles, but regardless, this is a VSP call and our work should align with the patterns that they've defined.

Reply:
I completely agree with you @ncksllvn, I did not agreed to use a custom function over a library but I am totally with all of you on using a build in function.